Pricenode crashes on startup due to CoinMarketCap retired API

wiz opened the pull request:
### Background
CoinMarketCap isn't actively used as a pricenode provider, because the only coins that used it have been delisted.

### Issue
CoinMarketCap has retired their v1 API, and the resulting `HTTP 410 Gone` causes the pricenode to crash on startup

### Summary
This PR simply removes CoinMarketCap as a PriceNodeProvider since it's apparently no longer used and the API has been retired

wiz committed:
Remove CoinMarketCap as dead code

sqrrm commented:
From reading the comments on how the Providers are instantiated, this looks fine. I don't understand exactly how that's done though, would really prefer the review of @cbeams

cbeams committed:
Update Poloniex exchange rate provider @Order value

With the removal of CoinMarketCap as an exchange rate provider in the
prior commit, the @Order values of the remaining three provider
implementations are non-contiguous. CoinMarketCap was @Order(3) and with
it gone, the remaining order values became 1, 2, 4. This is not a
problem in practice, but could be confusing for future maintainers, so
this commit simply decrements the Poloniex implementation from 4 to 3 in
order to make the values contiguous once again.

This commit also fixes a typo in ExchangeRateProvider's Javadoc.

wiz commented:
Again please note that this PR is an emergency measure since currently pricenodes crash on startup. Obviously we need to consider implementing one of the new CMC authenticated / paid APIs to replace the API that has been retired

cbeams reviewed (approved):
utACK, see my additional review commit.
